src/OpenTelemetry/Metrics/Exemplar/Exemplar.cs

```diff
@@ -103,6 +104,12 @@ public double DoubleValue
     internal void Update<T>(in ExemplarMeasurement<T> measurement)
         where T : struct
     {
+        if (Interlocked.Exchange(ref this.isCriticalSectionOccupied, 1) != 0)
+        {
+            // Some other thread is already writing, abort.
```

Comment:
It would be good to add a comment explaining why this is ok to return and not spind and wait to acquire the critical section.

Reply:
Added more detail to the comment